lib/process-base.js

```diff
@@ -0,0 +1,29 @@
+module.exports = class ProcessBase {
+    constructor(pid) {
+        if (this.constructor === ProcessBase) {
+            throw new Error('Cannot instantiate abstract class');
+        }
+
+        this.pid = pid;
+    }
+
+    attach() {
+        throw new Error('Abstract method cannot be called');
+    }
+
+    detach() {
+        throw new Error('Abstract method cannot be called');
+    }
+
+    /* generator */ iterateMemoryRegions() {
+        throw new Error('Abstract method cannot be called');
+    }
+
+    readMemory(address, size) {
+        throw new Error('Abstract method cannot be called');
+    }
+
+    writeMemory(address, buffer) {
```

Comment:
I was thinking on anything that can be indexed, like `Array` or `Uint8Array` (and maybe `ArrayBuffer`?).
`buffer` is just the name, I didn't mean node's `Buffer`, although it might still work.